djstripe/models.py

```diff
@@ -501,26 +500,29 @@ def subscribe(self, plan, quantity=1, trial_days=None,
         Trial_days corresponds to the value specified by the selected plan
         for the key trial_period_days.
         """
-        if ("trial_period_days" in djstripe_settings.PAYMENTS_PLANS[plan]):
-            trial_days = djstripe_settings.PAYMENTS_PLANS[plan]["trial_period_days"]
+
+        plan_instance = Plan.objects.get(stripe_id=plan)
+
+        if plan_instance.trial_period_days:
+            trial_days = plan_instance.trial_period_days
 
         if trial_days:
             resp = cu.update_subscription(
-                plan=djstripe_settings.PAYMENTS_PLANS[plan]["stripe_plan_id"],
+                plan=plan_instance,
                 trial_end=timezone.now() + datetime.timedelta(days=trial_days),
                 prorate=prorate,
                 quantity=quantity
             )
         else:
             resp = cu.update_subscription(
-                plan=djstripe_settings.PAYMENTS_PLANS[plan]["stripe_plan_id"],
+                plan=plan_instance,
                 prorate=prorate,
                 quantity=quantity
             )
         self.sync_current_subscription()
         if charge_immediately:
             self.send_invoice()
-        subscription_made.send(sender=self, plan=plan, stripe_response=resp)
+        subscription_made.send(sender=self, plan=plan_instance, stripe_response=resp)
```

Comment:
This line will break people's code if they're expecting an ID. I'd stick with returning the plan stripe id.